src/node/plugin.ts

```diff
@@ -225,12 +256,28 @@ export class PluginAPI {
 
     p.init({
       logger: logger,
+      workingDirectory: this.workingDirectory,
     })
 
     logger.debug("loaded")
 
     return p
   }
+
+  public async dispose(): Promise<void> {
+    await Promise.all(
+      Array.from(this.plugins.values()).map(async (p) => {
+        if (!p.deinit) {
+          return
+        }
+        try {
+          await p.deinit()
+        } catch (error) {
+          this.logger.error("plugin failed to deinit", field("name", p.name), field("error", error.message))
```

Comment:
Is it safe to assume `.message` exists on `error`?

Reply:
Nope lol

I'll run it through `logError` instead.

I do wish we could enforce the throw type with TypeScript though.

Reply:
> I do wish we could enforce the throw type with TypeScript though.

Is this possible?

Reply:
Nope, not yet at least. https://github.com/microsoft/TypeScript/issues/13219

Reply:
And since anything anywhere can throw whatever it wants I'm not sure it'll even help that much lol